sl_client_grpc.cc

```diff
@@ -0,0 +1,81 @@
+#include "sl_client_grpc.h"
+
+ServiceLayerClient::ServiceLayerClient() {
+  std::shared_ptr<grpc::Channel> channel = grpc::CreateChannel(
+      "localhost:50002", grpc::InsecureChannelCredentials());
+  stub_ = chirp::ServiceLayer::NewStub(channel);
+}
+
+ServiceLayerClient::~ServiceLayerClient() {}
+
+  std::string ServiceLayerClient::RegisterUser(const std::string& username) {
+  // set the fields of the request to pass to ServiceLayerServer
+  chirp::RegisterRequest request;
+  request.set_username(username);
+  chirp::RegisterReply reply;
+  grpc::ClientContext context;
+  // call ServiceLayerServer's registeruser function
+  grpc::Status status = stub_->registeruser(&context, request, &reply);
+  return "success";
+}
+
+void ServiceLayerClient::Chirp(const std::string& username, const std::string& text, const std::string& parent_id) {
+  // set the fields of the request to pass to ServiceLayerServer
+  chirp::ChirpRequest request;
+  request.set_username(username);
+  request.set_text(text);
+  request.set_parent_id(parent_id);
+  chirp::ChirpReply reply;
+  grpc::ClientContext context;
+  // call ServiceLayerServer's chirp function
+  grpc::Status status = stub_->chirp(&context, request, &reply);
+}
+
+void ServiceLayerClient::Follow(const std::string& username, const std::string& to_follow) {
+  // set the fields of the request to pass to ServiceLayerServer
+  chirp::FollowRequest request;
+  request.set_username(username);
+  request.set_to_follow(to_follow);
+
+  chirp::FollowReply reply;
+
+  grpc::ClientContext context;
+  // call ServiceLayerServer's follow function
+  grpc::Status status = stub_->follow(&context, request, &reply);
```

Comment:
you should probably return a bool or something to tell us if it was successful. For example, what if the person to_follow does not exit?

Reply:
Same for chirp function